library/packages/src/lib/y2packager/license.rb

```diff
@@ -12,40 +12,99 @@
 
 require "yast"
 require "digest"
+require "y2packager/licenses_fetchers"
 
 module Y2Packager
   # Represent a License which could be the same for multiple products.
   #
-  # This class stores the license ID and the traslated content of the license
-  # for different languages.
+  # This class represents a license.
   class License
     DEFAULT_LANG = "en_US".freeze
-    # @return [String] License unique identifier
-    attr_reader :id
 
     # @return [Boolean] whether the license has been accepted or not
     attr_reader :accepted
 
     # @return [Hash<String, String>] language -> content
     attr_reader :translations
 
+    # @return [Y2Packager::LicensesFetchers::Base] License fetcher object
+    attr_reader :fetcher
+
     alias_method :accepted?, :accepted
 
+    class << self
+      # Find a license for a given product
+      #
+      # @param product_name [String]    Product's name
+      # @param source       [:rpm,:url] Source to get the license from. For the time being,
+      #   only :rpm is really supported.
+      # @return [License]
+      def find(product_name, source, options = {})
+        return cache[product_name] if cache[product_name]
+
+        # This could be done in the constructor.
+        fetcher = LicensesFetchers.for(source, product_name, options)
+        new_license = License.new(fetcher)
+        return unless new_license.id
+
+        eq_license = cache.values.find { |l| l.id == new_license.id }
+        license = eq_license || new_license
+        cache[product_name] = license
+      end
+
+      def clear_cache
+        @cache = nil
+      end
+
+      private def cache
+        @cache ||= {}
+      end
+    end
+
     # Constructor
     #
     # @param options [Hash<String, String>]
-    def initialize(options = {})
+    def initialize(fetcher)
```

Comment:
Of course I'm also polluted with previous implementation, but I would expect to be able to instantiate a new License with its own content directly instead of looking for it. So would prefer named arguments like fetcher and content. Then the dummy fetcher could be used if a fetcher is not given with the given content